src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -872,5 +882,31 @@ private static NodeView GetNotConnectedNode(Window stepMainWindow, string nodeTy
             return byOriginUserAdded;
         }
 
+
+        internal static void SubscribeRunButtonNextStep(Step stepInfo, StepUIAutomation uiAutomationData, bool enableFunction, GuideFlow currentFlow)
+        {
+            CurrentExecutingStep = stepInfo;
+            if (uiAutomationData.Parameters.Any() && uiAutomationData.Parameters[0] is string)
+            {
+                Button runButton = GuideUtilities.FindChildInVisualTree(stepInfo.MainWindow, uiAutomationData.Parameters[0].ToString()) as Button;
+
+                if(runButton != null)
+                {
+                    if (enableFunction)
+                    {
+                        runButton.Click += RunButton_Click;
+                    }
+                    else
+                    {
+                        runButton.Click -= RunButton_Click;
+                    }
+                }
+            }
+        }
+
+        private static void RunButton_Click(object sender, RoutedEventArgs e)
```

Comment:
I just found that we can pass a complete class as argument to event handlers when you can pass the complete Step or the UIAutomation instance if is needed (so there is no need to use CurrentExecutingStep.UIAutomation[2]) , so you can use this feature instead of using CurrentExecutingGuide, see this example of passing a complete class as parameter, just check if is useful and also I there is no problem during the unsubscribe line.
![image](https://user-images.githubusercontent.com/61755417/167709698-17b4c599-a2d7-4820-8b19-49e6a12d33b2.png)